libsigner/src/events.rs

```diff
@@ -94,14 +76,32 @@ impl StacksMessageCodec for BlockProposalSigners {
         let block = NakamotoBlock::consensus_deserialize(fd)?;
         let burn_height = u64::consensus_deserialize(fd)?;
         let reward_cycle = u64::consensus_deserialize(fd)?;
-        Ok(BlockProposalSigners {
+        Ok(BlockProposal {
             block,
             burn_height,
             reward_cycle,
         })
     }
 }
 
+/// Event enum for newly-arrived signer subscribed events
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum SignerEvent<T: StacksMessageCodec + Clone> {
+    /// A miner sent a message over .miners
+    /// The `Vec<T>` will contain any signer messages made by the miner.
+    /// The `Option<StacksPublicKey>` will contain the message sender's public key if the vec is non-empty.
+    MinerMessages(Vec<T>, Option<StacksPublicKey>),
```

Comment:
A slightly cleaner way to do this would be to have `MinerMessasges(Vec<T>, StacksPublicKey)`, since it really ought not to be possible to send a `SignerEvent::MinerMessages(..)` with zero events.  What do you think?